### PR TITLE
solution react_todo-app-with-api task

### PR DESCRIPTION
**Pull request by ivaZaiets**
[DEMO LINK](https://ivaZaiets.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by tetyana-pol — APPROVED** [review]
Good job, but
in general, I think you should have used the Context, then you would have had less props

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,264 @@
...
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { Header } from './components/Header/Header';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isTodoSelected, setIsTodoSelected] = useState('');
```
Comment (polosanya) [inline src/App.tsx:21]: The name is a bit confusing. Sounds like a boolean

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,264 @@
...
+
+  const [isTodoLoading, setIsTodoLoading] = useState<Todo | null>(null);
```
Comment (polosanya) [inline src/App.tsx:23]: Fim similar cases
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,264 @@
...
+        setErrorMessage('Unable to update a todo');
+        throw error;
+      })
+      .finally(() => setIsTodoLoading(null));
+  };
+
+  let copyOfTodos = [...todos];
+
+  switch (isTodoSelected) {
+    case 'all':
```
Comment (polosanya) [inline src/App.tsx:200]: Consider using Enum instead of the hardcoded values

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,264 @@
...
+        });
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to update a todo');
+        throw error;
+      })
+      .finally(() => setIsTodoLoading(null));
+  };
+
+  let copyOfTodos = [...todos];
```
Comment (polosanya) [inline src/App.tsx:197]: Filter is a non-mutating method, so there is actually no need to create a copy

File: src/components/TodoError/TodoError.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  errorMessage: string,
+};
+
+export const TodoError: React.FC<Props> = ({
+  onErrorMessage,
+  errorMessage,
+}) => {
+  return (
+    <>
+      <div
```
Comment (polosanya) [inline src/components/TodoError/TodoError.tsx:16]: React fragment is redundant here as you already have a container (div)

**Review by polosanya — APPROVED** [review]
Good job! Left a few comments that might be useful)

**Pull request closed without merging**